dpnp/dpnp_iface_linearalgebra.py

```diff
@@ -274,58 +290,328 @@ def matmul(x1, x2, out=None, **kwargs):
     >>> c = np.ones([9, 5, 4, 3])
     >>> np.matmul(a, c).shape
```

Comment:
This behavior is demonstrable in the third example (`>>> a = np.arange(2 * 2 * 4).reshape((2, 2, 4))`) so it can be removed